inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -85,16 +91,112 @@ class XmlSerializer : public ngraph::AttributeVisitor {
 public:
     XmlSerializer(pugi::xml_node& data,
                   std::ostream& bin_data,
-                  std::string& node_type_name)
+                  std::string& node_type_name,
+                  const std::map<std::string, ngraph::OpSet>& custom_opsets)
         : m_xml_node(data)
         , m_bin_data(bin_data)
-        , m_node_type_name(node_type_name) {
+        , m_node_type_name(node_type_name)
+        , m_custom_opsets(custom_opsets) {
+    }
+
+    std::vector<std::string> map_type_from_body(const pugi::xml_node& xml_node,
+        const std::string& map_type) {
+        std::vector<std::string> output;
+        if (map_type == "Result" || map_type == "Parameter") {
+            for (pugi::xml_node node : xml_node.child("body").child("layers")) {
+                auto param = map_type.compare(node.attribute("type").value());
+                if (!param) {
+                    output.push_back(node.attribute("id").value());
+                }
+            }
+
+            // parameters/results for serialized body function are provided in reversed order
+            std::reverse(output.begin(), output.end());
+        }
+
+        return output;
     }
 
     void on_adapter(const std::string& name,
                     ngraph::ValueAccessor<void>& adapter) override {
         (void)name;
-        (void)adapter;
+
+        if (m_xml_node.parent().child("body")) {
```

Comment:
This comment is about potential refactoring - I don't expect to do it in this PR but we could take a note to do this after release when we'll have more time.

This if is too complicated and because of it it's hard to understand.  I suggest to extract 2 separate helper functions and have something like this:

```
on_adapter(const std::string& name, ngraph::ValueAccessor<void>& adapter)
{
    if (m_xml_node.parent().child("body"))
    {
        if (name == "input_descriptions") 
            input_descriptions_on_adapter();
        else if (name == "output_descriptions")  
            output_descriptions_on_adapter();                
    }
}
```

Reply:
I have created a jira ticket (46975) for all future improvements to not forget about it. This issue is already addressed as a subtask (46976).